### ISSUE TITLE
current repl maybe cannot recognize mathcal letter?

### ISSUE THREAD

**Issue by ohyeat** [leanprover-community/repl#28]
When I was testing repl using the following example (it's actually the predefined theorem ```Metric.uniformity_basis_dist``` in mathlib4 ```Mathlib/Topology/MetricSpace/PseudoMetric.lean```):

```bash
import Mathlib
open Set Filter TopologicalSpace Bornology
open scoped BigOperators ENNReal NNReal Uniformity Topology
universe u v w
variable {α : Type u} {β : Type v} {X ι : Type*}
variable [PseudoMetricSpace α]
open Lean Meta Qq Function
variable {x y z : α} {δ ε ε₁ ε₂ : ℝ} {s : Set α}
open Metric

example {r : ℝ} (h0 : 0 < r) (h1 : r < 1) :
    (𝓤 α).HasBasis (fun _ : ℕ => True) fun n : ℕ => { p : α × α | dist p.1 p.2 < r ^ n } :=
  Metric.mk_uniformity_basis (fun _ _ => pow_pos h0 _) fun _ε ε0 =>
    let ⟨n, hn⟩ := exists_pow_lt_of_lt_one ε0 h1
    ⟨n, trivial, hn.le⟩
```
The above proof gets no warning or error message in interaction mode in vscode. 

But when I test them via repl, I got the following feedback
```bash
> lake exe repl
{"cmd": "import Mathlib\nopen Set Filter TopologicalSpace Bornology\nopen scoped BigOperators ENNReal NNReal Uniformity Topology\nuniverse u v w\nvariable {\u03b1 : Type u} {\u03b2 : Type v} {X \u03b9 : Type*}\nvariable [PseudoMetricSpace \u03b1]\nopen Lean Meta Qq Function\nvariable {x y z : \u03b1} {\u03b4 \u03b5 \u03b5\u2081 \u03b5\u2082 : \u211d} {s : Set \u03b1}\nopen Metric\n"}

{"env": 0}
{"cmd": "example :\n    (\ud835\udce4 \u03b1).HasBasis (fun \u03b5 : \u211d => 0 < \u03b5) fun \u03b5 => { p : \u03b1 \u00d7 \u03b1 | dist p.1 p.2 < \u03b5 } := by\n  rw [toUniformSpace_eq]\n  exact UniformSpace.hasBasis_ofFun (exists_gt _) _ _ _ _ _\n", "env": 0}

{"messages":
 [{"severity": "error",
   "pos": {"line": 2, "column": 5},
   "endPos": null,
   "data": "expected token"}],
 "env": 1}
```
The "pos" in error message indicates that the error occurred due to the mathcal character 𝓤. I tried another example

```bash
import Mathlib
open Set Filter TopologicalSpace Bornology
open scoped BigOperators ENNReal NNReal Uniformity Topology
universe u v w
variable {α : Type u} {β : Type v} {X ι : Type*}
variable [PseudoMetricSpace α]
open Lean Meta Qq Function
variable {x y z : α} {δ ε ε₁ ε₂ : ℝ} {s : Set α}
open Metric

example {p : α → Prop} :
    (∀ᶠ y in 𝓝 x, p y) ↔ ∃ ε > 0, ∀ ⦃y⦄, dist y x < ε → p y :=
  mem_nhds_iff
```
and got the same error message in repl

```bash
> lake exe repl
{"cmd": "import Mathlib\nopen Set Filter TopologicalSpace Bornology\nopen scoped BigOperators ENNReal NNReal Uniformity Topology\nuniverse u v w\nvariable {\u03b1 : Type u} {\u03b2 : Type v} {X \u03b9 : Type*}\nvariable [PseudoMetricSpace \u03b1]\nopen Lean Meta Qq Function\nvariable {x y z : \u03b1} {\u03b4 \u03b5 \u03b5\u2081 \u03b5\u2082 : \u211d} {s : Set \u03b1}\nopen Metric\n"}

{"env": 0}
{"cmd": "example {p : \u03b1 \u2192 Prop} :\n    (\u2200\u1da0 y in \ud835\udcdd x, p y) \u2194 \u2203 \u03b5 > 0, \u2200 \u2983y\u2984, dist y x < \u03b5 \u2192 p y :=\n  mem_nhds_iff\n", "env": 0}

{"messages":
 [{"severity": "error",
   "pos": {"line": 2, "column": 13},
   "endPos": null,
   "data": "expected token"}],
 "env": 1}
```
The error message also indicates the error occurred due to a mathcal letter '𝓝', therefore I guess the issue is about the unicode transformation on mathcal letter that the repl cannot handle. 

If my guess is correct? How can I solve this issue? Thanks!

**Comment by ohyeat** [comment]
I solved the issue. It occurred when I encoded the string before feed into repl using python function ```json.dumps``` without setting ```ensure_ascii=False```.

**Comment by karlcuinju** [comment]
I also encountered a similar problem，like the figure below. And the error message is due to '**≤**'.
<img width="1039" alt="image" src="https://github.com/leanprover-community/repl/assets/115494933/c2ba346b-b34b-4e2c-a421-fe7d9eaae2b9">

Seems you have solved the issues, what is your input format of repl finally？

**Comment by ohyeat** [comment]
> I also encountered a similar problem，like the figure below. And the error message is due to '**≤**'. <img alt="image" width="1039" src="https://private-user-images.githubusercontent.com/115494933/341333888-c2ba346b-b34b-4e2c-a421-fe7d9eaae2b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4NzIzOTgsIm5iZiI6MTcxODg3MjA5OCwicGF0aCI6Ii8xMTU0OTQ5MzMvMzQxMzMzODg4LWMyYmEzNDZiLWIzNGItNGUyYy1hNDIxLWZlN2Q5ZWFhZTJiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyMFQwODI4MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNGIxOGExMjA1MTQ3MzZkMGU0ZjkwY2UzMDg4NDAwZDdmMTA2MjgyYTZhNTQzZGI0N2EzNjUxNzkzNmZmZjQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KhlhMBHnvQviW-gUkU1rTT276Y_GalE72kFWmn8B1tc">
> 
> Seems you have solved the issues, what is your input format of repl finally？

How did you feed the proof sketch into the repl? If you just copy the sketch into repl via terminal, will you meet the same issue?

**Comment by karlcuinju** [comment]
> > I also encountered a similar problem，like the figure below. And the error message is due to '**≤**'. <img alt="image" width="1039" src="https://private-user-images.githubusercontent.com/115494933/341333888-c2ba346b-b34b-4e2c-a421-fe7d9eaae2b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4NzIzOTgsIm5iZiI6MTcxODg3MjA5OCwicGF0aCI6Ii8xMTU0OTQ5MzMvMzQxMzMzODg4LWMyYmEzNDZiLWIzNGItNGUyYy1hNDIxLWZlN2Q5ZWFhZTJiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyMFQwODI4MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNGIxOGExMjA1MTQ3MzZkMGU0ZjkwY2UzMDg4NDAwZDdmMTA2MjgyYTZhNTQzZGI0N2EzNjUxNzkzNmZmZjQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KhlhMBHnvQviW-gUkU1rTT276Y_GalE72kFWmn8B1tc">
> > Seems you have solved the issues, what is your input format of repl finally？
> 
> How did you feed the proof sketch into the repl? If you just copy the sketch into repl via terminal, will you meet the same issue?

Yes, I just copy the sketch into repl via terimal, then encountered the issue.

**Comment by ohyeat** [comment]
> > > I also encountered a similar problem，like the figure below. And the error message is due to '**≤**'. <img alt="image" width="1039" src="https://private-user-images.githubusercontent.com/115494933/341333888-c2ba346b-b34b-4e2c-a421-fe7d9eaae2b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4NzIzOTgsIm5iZiI6MTcxODg3MjA5OCwicGF0aCI6Ii8xMTU0OTQ5MzMvMzQxMzMzODg4LWMyYmEzNDZiLWIzNGItNGUyYy1hNDIxLWZlN2Q5ZWFhZTJiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyMFQwODI4MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNGIxOGExMjA1MTQ3MzZkMGU0ZjkwY2UzMDg4NDAwZDdmMTA2MjgyYTZhNTQzZGI0N2EzNjUxNzkzNmZmZjQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KhlhMBHnvQviW-gUkU1rTT276Y_GalE72kFWmn8B1tc">
> > > Seems you have solved the issues, what is your input format of repl finally？
> > 
> > 
> > How did you feed the proof sketch into the repl? If you just copy the sketch into repl via terminal, will you meet the same issue?
> 
> Yes, I just copy the sketch into repl via terimal, then encountered the issue.

<img width="1090" alt="temp" src="https://github.com/leanprover-community/repl/assets/9134351/8e193e9f-5540-4606-b88d-367394b81a3d">
 No idea why you can't, maybe you can check the input method on your computer.

**Comment by karlcuinju** [comment]
> > > > I also encountered a similar problem，like the figure below. And the error message is due to '**≤**'. <img alt="image" width="1039" src="https://private-user-images.githubusercontent.com/115494933/341333888-c2ba346b-b34b-4e2c-a421-fe7d9eaae2b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4NzIzOTgsIm5iZiI6MTcxODg3MjA5OCwicGF0aCI6Ii8xMTU0OTQ5MzMvMzQxMzMzODg4LWMyYmEzNDZiLWIzNGItNGUyYy1hNDIxLWZlN2Q5ZWFhZTJiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyMFQwODI4MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNGIxOGExMjA1MTQ3MzZkMGU0ZjkwY2UzMDg4NDAwZDdmMTA2MjgyYTZhNTQzZGI0N2EzNjUxNzkzNmZmZjQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KhlhMBHnvQviW-gUkU1rTT276Y_GalE72kFWmn8B1tc">
> > > > Seems you have solved the issues, what is your input format of repl finally？
> > > 
> > > 
> > > How did you feed the proof sketch into the repl? If you just copy the sketch into repl via terminal, will you meet the same issue?
> > 
> > 
> > Yes, I just copy the sketch into repl via terimal, then encountered the issue.
> 
> <img alt="temp" width="1090" src="https://private-user-images.githubusercontent.com/9134351/341341684-8e193e9f-5540-4606-b88d-367394b81a3d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4NzQyOTcsIm5iZiI6MTcxODg3Mzk5NywicGF0aCI6Ii85MTM0MzUxLzM0MTM0MTY4NC04ZTE5M2U5Zi01NTQwLTQ2MDYtYjg4ZC0zNjczOTRiODFhM2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjBUMDg1OTU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmRlMzM5YWRlNTcwZjg1ZGE5YjBjYWYxNTczMjA1YTNlNzYzNzJiNTEyYzdhM2VkMDJjZDZhNjc1OTJjOTU1OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.TR7XRchJ85Hu5MlhReYw7niSMXlu0p5zY60n5JnGabQ"> No idea why you can't, maybe you can check the input method on your computer.

Thanks a lot. I am new to Lean and Lake, I want to know that did you change the settings of the repo, like "lakefile.lean, lake-manifest.json" files, or just clone the repo and run "lake exe repl"?

**Comment by ohyeat** [comment]
Nope, I just followed README. I think your repl works normally, you can check your proof sketch in user interactive mode in vscode.

**Comment by sorgfresser** [comment]
As a follow up: I faced a similar issue with imports, for me it was a version mismatch in the lean toolchain between the lake environment I was running the repl from and the version that was used to build the lean repl.
